### ISSUE TITLE
Validator discussion

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor#2892]
The purpose of this issue is to discuss the architecture for an AsciiDoc validator and the rules it should/could enforce.

The validator will be a separate tool. However, until it exists, it's helpful to have a place to discuss its requirements.

**Comment by elextr** [comment]
Good idea.  Do you see it purely as a validator of Asciidoc syntax (in which case its likely fairly short), or also as a linter picking up doubtful syntax?

I would also comment that it will need some care either not to include backend specific rules in the validator or to clearly identify them as such, so they can be controlled by an option to specify which backend.
